pipeline/lib/storage.ml

```diff
@@ -26,6 +27,7 @@ let setup_metadata ~repository (db : Postgresql.connection) =
   in
   try
     let result = db#exec query in
+    db#finish;
```

Comment:
The `#finish` is a bit early before the `#get_all`! Also it needs to happen even in case of an exception :)

Reply:
Maybe we should introduce a helper function:

```ocaml
let with_db ~conninfo fn =
  let db = new Postgresql.conn ... in
  match fn db with
  | v -> db#finish ; v
  | exception e ->
       Logs.err (...) ;
       db#finish ;
       raise exn
```